core-container/entry.sh

```diff
@@ -43,56 +33,78 @@ function set_provider_adapters() {
     rm kcpsecret
 
     echo "Delete the kcp provider object if it already exists, and then create it."
-    if ! kubectl delete spaceproviderdesc pkcp ; then
+    if ! kubectl delete spaceproviderdesc $PROVIDER_NAME ; then
```

Comment:
Why {delete; create} instead of {apply}? The delete;create generates a disturbance for any process monitoring that object.

Reply:
I do not know what you mean by "global restart".

In the MCCM code we have not yet attempted to support in-place upgrades. Are you supporting them in the Space Framework? If so, what rules have you established for what upgrades may be done in place vs. what may not? For how to roll an upgrade into a running distributed system?

The usual pattern for this is to first expand what consumers can consume, and then expand what producers can produce. When there is a data store it gets more interesting because that plays both roles --- you have more sequencing: first expand what the data store can store but DO NOT YET expand what it actually stores; then expand what other consumers can consume; then expand what other producers actually produce. Doing `delete;create` instead of `apply` does not accomplish these things.